instrumentation/opentelemetry-instrumentation-starlette/src/opentelemetry/instrumentation/starlette/__init__.py

```diff
@@ -331,9 +332,6 @@ def __init__(self, *args: Any, **kwargs: Any):
         # adding apps to set for uninstrumenting
         _InstrumentedStarlette._instrumented_starlette_apps.add(self)
 
-    def __del__(self):
-        _InstrumentedStarlette._instrumented_starlette_apps.remove(self)
```

Comment:
This would fail if the app isn't in the set, which would happen in the `.clear()` call in `StarletteInstrumentation.uninstrument()`. And actually, `__del__` could only ever be called after such a call because the set held a strong reference, preventing `__del__` from being called. I changed to `WeakSet` which should fix it, and we don't need `__del__` since it's automatically cleaned when the object goes away